pipeline.py

```diff
@@ -0,0 +1,322 @@
+
+def main():
+    backend = hb.ServiceBackend(
+        billing_project=BILLING_PROJECT,
+        remote_tmpdir=TMP_DIR,
+    )
+    b = hb.Batch(
+        name=f"Decontam-Mixing-Cram-Files",
+        requester_pays_project='daly-ibd',
+        default_python_image='us-central1-docker.pkg.dev/broad-mpg-gnomad/wlu/hail/hail-pysam-samtools:latest',
+        backend=backend,
+    )
+
+
+    logger.info('Loading sample info...')
+    sample_paths = pd.read_csv(args.input_sample_info, sep=',')
+    sample_ids = pd.read_csv(args.selected_samples, sep=',')
+    sample_paths = sample_paths[sample_paths['s'].isin(sample_ids['hgdp_id'])]
+    samples = []
+    cram_files = {}
+    gvcf_files = {}
+    sample_pops = {}
+    for sample in sample_paths.iterrows():
+        samples.append(sample[1][0])
+        cram_files[sample[1][0]] = (f'{sample[1][1]}', f'{sample[1][1]}.crai')
+        gvcf_files[sample[1][0]] = (f'{sample[1][2]}', f'{sample[1][2]}.tbi')
+        sample_pops[sample[1][0]]= sample[1][3]
+
+    input_ref_fasta = b.read_input_group(
+        **{"fasta": REF_FASTA_PATH,
+           "fasta.fai": REF_FASTA_INDEX,
+           "dict":REF_DICT}
+    )
+
+    contam_free_gvcfs = {}
+    contam_free_crams = {}
+    for s in samples:
+        sample_id = s
+        contam_free_crams[sample_id] = {}
+        contam_free_gvcfs[sample_id] = {}
+        logger.info(f'-------Decontaminating {sample_id}-------')
+        input_cram_file = b.read_input_group(
+            **{"cram": cram_files[s][0], "cram.crai": cram_files[s][1]}
+        )
+        if not args.skip_gvcf_dict:
+            logger.info(f'Generating gvcf dict - {sample_id}...')
+            gvcf_path = gvcf_files[s]
+            gvcf_file_name = gvcf_path[0].split('/')[-1][:-7]
+            gvcf_dict = run_gvcf_dict(b, gvcf_file_name, gvcf_path)
+
+        cram_job_depend_list = []
+        gvcf_job_depend_list = []
+        for chromosome in chromosomes:
+            chromosome = 'chr21'
+            output_cram_path = f"{MY_BUCKET}/contam_free/{sample_id}/cram_by_chrom/{sample_id}_contam_free_{chromosome}.cram"
+            logger.info(f'Generating contamination free cram: {sample_id}-{chromosome}...')
+            depend_on = None
+            if not hl.hadoop_exists(output_cram_path):
+                j_cram = b.new_python_job(name=f"Run_{sample_id}_{chromosome}_contam_free_file")
+                j_cram._machine_type = "n1-highmem-16"
+                j_cram.storage("500Gi")
+                j_cram.call(
+                    write_contam_free_cram_file,
+                    gvcf_dict,
+                    input_cram_file,
+                    input_ref_fasta,
+                    j_cram.ofile,
+                    chromosome
+                )
+
+                logger.info(f'Indexing contamination free cram: {sample_id}-{chromosome}...')
+                j_index = b.new_job(name=f"Index_{sample_id}_{chromosome}_contam_free_file")
+                depend_on = j_index
```

Comment:
legacy code?

Reply:
> legacy code?

I don't think it is. but do you think this is unnecessary in some way?